File: HW06_SNS_SQS/Node/src/snsPublisher.js
require('dotenv').config();

var AWS = require('aws-sdk');
AWS.config.update({ region: `${process.env.REGION}` });

const LOW = 0; 
const HIGH = 100;
const DELAY = 5000;
const MAX = 1000;
let randomNumber = 0;
let counter = 1;

function publish() {
  setTimeout(function() {
    randomNumber = LOW + Math.floor(Math.random() * (HIGH - LOW)) + 1;
    console.log(`${counter}: ${randomNumber}`);

    let params = {
      Message: `${randomNumber}`,
      TopicArn: `${process.env.TOPIC_ARN}`
    };

    var publishTextPromise = new AWS.SNS({ apiVersion: '2010-03-31' }).publish(params).promise();

    publishTextPromise.then(
      function(data) {
        counter++;
        console.log(`Message ${params.Message} sent to the topic ${params.TopicArn}`);
        console.log("MessageID is " + data.MessageId);
      }).catch(
        function(err) {
          console.error(err, err.stack);
        });

    if (counter < MAX) {
      publish();
    }
  }, DELAY)
}
publish();

File: HW06_SNS_SQS/Python/src/SNSPublisher.py
import boto3
import random
import time

boto3.setup_default_session(profile_name='YOUR_PROFILE_NAME')

LOW = 1
HIGH = 100
DELAY = 5
MAX = 1000
randomNumber = 0
counter = 0
topicArn = 'arn:aws:sns:YOUR_REGION:YOUR_TOPIC_ARN'

sns = boto3.client('sns', region_name='YOUR_REGION')

while counter < MAX:
    randomNumber = random.randrange(LOW, HIGH)

    message = sns.publish(
        TopicArn=topicArn,
        Message=str(randomNumber),
    )

    counter += 1
    print(str(counter) + ". Message " + str(randomNumber) + " sent to the topic " + topicArn)

    time.sleep(DELAY)
